packages-exp/app-check-exp/src/providers.ts

```diff
@@ -0,0 +1,123 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { FirebaseApp, _getProvider } from '@firebase/app-exp';
+import { Provider } from '@firebase/component';
+import { issuedAtTime } from '@firebase/util';
+import { exchangeToken, getExchangeRecaptchaTokenRequest } from './client';
+import { AppCheckError, ERROR_FACTORY } from './errors';
+import { CustomProviderOptions } from './public-types';
+import {
+  getToken as getReCAPTCHAToken,
+  initialize as initializeRecaptcha
+} from './recaptcha';
+import { AppCheckProvider, AppCheckTokenInternal } from './types';
+
+/**
+ * App Check provider that can obtain a reCAPTCHA V3 token and exchange it
+ * for an AppCheck token.
```

Comment:
As mentioned above, is there a reason to use literal-looking "AppCheck" instead of "App Check?"

Reply:
Changed to product name.